src/Diagnostics.DataProviders/DataProviderLogDecorator.cs

```diff
@@ -68,6 +68,11 @@ public Task<DataTable> ExecuteClusterQuery(string query, string requestId = null
             return ExecuteQuery(query, DataProviderConstants.FakeStampForAnalyticsCluster, requestId, operationName);
         }
 
+        public Task<KustoQuery> GetKustoQuery(string query, string stampName)
```

Comment:
should we add one more for GetKustoClusterQuery(string query) , so no stamp name as input

Reply:
I didn't even knew that we had a method call ExecuteClusterQuery. Your ask maskes sense so pushing a commit